rest_framework/fields.py

```diff
@@ -576,14 +576,18 @@ def __init__(self, **kwargs):
             self.validators.append(MinLengthValidator(self.min_length, message=message))
 
     def run_validation(self, data=empty):
+        (is_empty_value, data) = self.validate_empty_values(data)
+        if is_empty_value:
+            return data
+
+        value = self.to_internal_value(data)
```

Comment:
I would consider re-jigging how this works - as it is this kinda confuses me.

Could we just alter the `if data == '':` check so that it also checks for whitespace only. Eg: `if data == '' or self.trim_whitespace and six.text_type(data).strip() == '':`